tls/s2n_resume.c

```diff
@@ -470,13 +470,15 @@ int s2n_connection_get_session(struct s2n_connection *conn, uint8_t *session, si
     POSIX_ENSURE_REF(conn);
     POSIX_ENSURE_REF(session);
 
-    int len = s2n_connection_get_session_length(conn);
+    const int len = s2n_connection_get_session_length(conn);
+    POSIX_GUARD(len);
+    const size_t size = len;
```

Comment:
It seems kind of odd to have two variables that are essentially the same. I think there were a couple of other similar places in this PR as well. It's probably fine here and in most of these cases, but in general it seems like there's a chance it could cause confusion about which variable is supposed to be used. I wonder if standardizing the naming could help? Like, calling the return value `len_ret`, and then setting `len` with the correct type after it's checked? That way it's more clear which is supposed to be actually used. Not sure though, maybe this way is okay.

Reply:
Actually, wouldn't another option be to just cast in the POSIX_ENSURE and not make a new variable?